Admin/AdminInterface.php

```diff
@@ -656,6 +657,21 @@ public function delete($object);
     /**
      * @param mixed $object
      *
+     * @return int|null
+     */
+    public function getVersion($object);
```

Comment:
@EmmanuelVella you can only have the catch in the ``CRUDController`` to set the correct error message. However the check can be done inside an extension. see: the ``AdminExtension`` class and the related documentation: https://sonata-project.org/bundles/admin/2-3/doc/reference/extensions.html